zircon.core/src/commonMain/kotlin/org/hexworks/zircon/api/resource/TilesetResource.kt

```diff
@@ -14,6 +14,12 @@ interface TilesetResource : Identifiable {
 
     val tileType: TileType
     val tilesetType: TilesetType
+    /**
+     * A "stringly-typed" field to identify tilesets to be loaded by custom tileset loaders.
+     *
+     * For example, a tileset loader designed for Tiled might use a subtype of `TILED`.
+     */
+    val tilesetSubtype: String? get() = null
```

Comment:
I wanted to ask "why not make it an `enum`?" then I realized that it wouldn't work because this is for custom code. 😄 Can we call this a `tilesetId`, or `key`, or something that implies that it should be unique? Does it need to be unique?

Reply:
> Does it need to be unique?

Kinda. At least with Tiled, the thing that creates the Tiled TilesetResource is my `zircon-tiled` library, and the thing that reads those resources is the tileset loader that's also in `zircon-tiled`. I expect it to be common that the same library/module is going to be responsible for reading in the raw data (TilesetResource) and for displaying it (TilesetLoader).

However, this isn't necessarily the case. Someone could create a level format standard in XML or JSON and someone else could create a library that displays it.

---

While "subtype" is _kinda_ vague, I'm having trouble coming up with anything better. Right now my Tiled loader looks like this:

```kotlin
class TiledSwingGraphicTilesetLoader : TilesetLoader<Graphics2D> {
    private val tilesetCache = mutableMapOf<UUID, Tileset<Graphics2D>>()

    override fun loadTilesetFrom(resource: TilesetResource): Tileset<Graphics2D> =
            tilesetCache.getOrPut(resource.id) { TiledMapTileset(resource) }

    override fun canLoadResource(resource: TilesetResource): Boolean =
            resource.tilesetSubtype == TiledTilesetResource.SUBTYPE
}
```
and notice that it's not even checking the tilesetType, which..._could_ be fine? If we were doing this from scratch, tilesetType would probably be a sealed class with that looked like this:

```kotlin
sealed class TilesetType {
    object CP437 : TilesetType()
    object TrueTypeFont : TilesetType()
    object Graphic : TilesetType()
    class Custom(val type: String) : TilesetType()
}
```

But it's a little late for that now.

Reply:
Yeah. Then this is fine. I'll look into whether we can do the `sealed class` solution without breaking the API.